packaging_automation/github_stats_collector.py

```diff
@@ -0,0 +1,111 @@
+import argparse
+from datetime import datetime
+from enum import Enum
+from typing import Dict, Any
+
+from github import Github
+from sqlalchemy import Column, DATE, INTEGER, TIMESTAMP, ForeignKey, String, UniqueConstraint
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker, relationship
+
+from .dbconfig import (Base, db_connection_string, DbParams,db_session)
+
+ORGANIZATION_NAME = "citusdata"
+
+
+class GithubRepos(Enum):
+    citus = "citus"
+    pg_auto_failover = "pg-auto-failover"
+
+
+class GithubCloneStatsTransactionsMain(Base):
+    __tablename__ = "github_stats_clone_transactions_main"
+    id = Column(INTEGER, primary_key=True, autoincrement=True)
+    fetch_time = Column(TIMESTAMP, nullable=False)
+    repo_name = Column(String, nullable=False)
+    count = Column(INTEGER, nullable=False)
+    uniques = Column(INTEGER, nullable=False)
+    details = relationship("GithubCloneStatsTransactionsDetail", backref="github_stats_clone_transactions_main",
+                           lazy=True)
+
+
+class GithubCloneStatsTransactionsDetail(Base):
+    __tablename__ = "github_stats_clone_transactions_detail"
+    id = Column(INTEGER, primary_key=True, autoincrement=True)
+    clone_date = Column(DATE, nullable=False)
+    count = Column(INTEGER, nullable=False)
+    uniques = Column(INTEGER, nullable=False)
+    parent_id = Column(INTEGER, ForeignKey('github_stats_clone_transactions_main.id'), nullable=False)
+
+
+class GithubCloneStats(Base):
+    __tablename__ = "github_clone_stats"
+    id = Column(INTEGER, primary_key=True, autoincrement=True)
+    repo_name = Column(String, nullable=False)
+    fetch_time = Column(TIMESTAMP, nullable=False)
+    clone_date = Column(DATE, nullable=False)
+    count = Column(INTEGER, nullable=False)
+    uniques = Column(INTEGER, nullable=False)
+    __table_args__ = (UniqueConstraint('repo_name', 'clone_date', name='repo_name_clone_date_uq'),)
+
+
+def stat_records_exists(record_time: datetime.date, session) -> bool:
+    db_record = session.query(GithubCloneStats).filter_by(clone_date=record_time).first()
+    return db_record is not None
+
+
+def github_clone_stats(github_token: str, organization_name: str, repo_name: str) -> Dict[str, Any]:
+    g = Github(github_token)
+    repo = g.get_repo(f"{organization_name}/{repo_name}")
+    return repo.get_clones_traffic(per="day")
+
+
+
+
+def fetch_and_store_github_clones(organization_name: str, repo_name: str, db_parameters: DbParams, github_token: str,
+                                  is_test: bool):
+    contents = github_clone_stats(github_token, organization_name, repo_name)
+
+    session = db_session(db_parameters, is_test)
+
+    fetch_time = datetime.now()
+
+    main_transaction = GithubCloneStatsTransactionsMain(fetch_time=fetch_time, count=contents['count'],
+                                                        repo_name=repo_name, uniques=contents['uniques'])
```

Comment:
I'm a bit confused here. Doesn't `get_clones_traffic` in `github_clone_stats` return a list of clone objects. Why would the contents include `count` and `unique`? Am I reading the docs wrong?